cpp/src/arrow/python/builtin_convert.cc

```diff
@@ -287,233 +319,217 @@ class TypedConverter : public SeqConverter {
   BuilderType* typed_builder_;
 };
 
-class BoolConverter : public TypedConverter<BooleanBuilder> {
+template <typename BuilderType>
+class TypedConverterVisitor : public TypedConverter<BuilderType> {
```

Comment:
I see that the base class here `SeqConverter` does not have a virtual destructor (which can cause a memory leak), can you add one while we're touching this code?

`virtual ~SeqConverter() {}`